src/MyConnection.h

```diff
@@ -18,10 +18,15 @@ class MyConnection : boost::noncopyable {
 
 public:
 
-  MyConnection(std::string host, std::string user, std::string password,
-               std::string db, unsigned int port, std::string unix_socket,
-               unsigned long client_flag, std::string groups,
-               std::string default_file) :
+  MyConnection(Rcpp::Nullable < std::string > host,
```

Comment:
I tested your RcppCore/Rcpp#417 up until krlmlr/Rcpp@2c57b98.

The test code compiles ok now:

    // with Rcpp PR 417
    void foo(const Rcpp::Nullable < Rcpp::CharacterVector > & myVar) {
      std::string s = Rcpp::as<std::string>(myVar);
    }
    // OK

The issue with spacing also seems to have disappeared:

    Generated extern "C" functions 
    --------------------------------------------------------
    #include <Rcpp.h>
    // foo
    void foo(const Rcpp::Nullable < Rcpp::CharacterVector >& myVar);
    RcppExport SEXP sourceCpp_0_foo(SEXP myVarSEXP) {
    BEGIN_RCPP
        Rcpp::RNGScope __rngScope;
        Rcpp::traits::input_parameter< const Rcpp::Nullable < Rcpp::CharacterVector >& >::type myVar(myVarSEXP);
        foo(myVar);
        return R_NilValue;
    END_RCPP
    }

My changes to the RMySQL `MyConnection.h`, `MyConnection` constructor 
now also compile with const reference parameters. Some quick tests show 
everything still works ok.

Should `connnection.cpp`, `connection_create` also have const reference 
parameters? I tried it and everything seems ok.

With regard to your question:

> Are you OK with leaving it "as is" until Rcpp is fixed?

Not sure what you mean by that:

- Leaving RMySQL master "as is", i.e. not applying this PR#90 until 
  Rcpp is fixed and `const&` can be added, or
- Pulling this PR#90 "as is", holding off the `const&` optimization 
  until Rcpp is fixed.

For myself, it doesn't matter much, because I'm working from a local 
branch that has my own changes incorporated anyway. However, for other 
users of RMySQL development master it might be nice to have the big 
problem solved now and do the optimizations later.

I do still need to look at the other issue you raised though, creating 
a function for that repeated pattern. I will try to get to that later 
this week.